R/MSnExperiment-class.R

```diff
@@ -477,19 +477,17 @@ readMSnExperiment <- function(file, sampleData, backend = BackendMzR(),
         hdr[order(hdr$acquisitionNum), ]
     }
     spectraData <- DataFrame(
-        do.call(rbind, bplapply(file, .read_file, files=file,
-                                smoothed=smoothed, BPPARAM=BPPARAM)))
-    backend <- backendInitialize(backend, file, spectraData, ...,
-                                 BPPARAM=BPPARAM)
-    backend <- backendImportData(backend, spectraData, BPPARAM=BPPARAM)
-    new("MSnExperiment",
-        backend = backend,
+        do.call(rbind, bplapply(file, .read_file, files = file,
+                                smoothed = smoothed, BPPARAM = BPPARAM)))
+    msnexp <- new("MSnExperiment",
+        backend = backendInitialize(BackendMzR(), file),
         sampleData = sampleData,
         spectraData = spectraData,
         processingQueue = list(),
         metadata = metadata,
         processing = paste0("Data loaded [", date(), "]")
     )
+    setBackend(msnexp, backend, ..., BPPARAM = BPPARAM)
```

Comment:
And what if we just use `setBackend` if the `backend` is not a. Sure, it adds another `if` condition, but I have a bad feeling with allowing `modCount` to be `1`. If you created an `MSnExperiment` with the constructor function you would get a `modCount = 0`.

Thus I would suggest:
```r
if (!inherits(backend, "BackendMzR"))
    setBackend(msnexp, backend, ..., BPPARAM = BPPARAM)
```

Reply:
That's better! I will change that.